### PR TITLE
Properly handle PAGE_NOACCESS and PAGE_GUARD

### PR DESCRIPTION
**Pull request by zefklop**
## Purpose

Properly handle page fault in protected areas for file-backed sections.

JIRA issue: [ROSTESTS-110](https://jira.reactos.org/browse/ROSTESTS-110)

## Proposed changes

The page fault handler should fail early in case of a region marked as PAGE_NOACCESS or PAGE_GUARD, otherwise it allocates a page as if handling it in PAGE_READWRITE, and then bugchecks afterwards because of this inconsistency.

### CODE REVIEW

File: ntoskrnl/mm/section.c [hunk 1]
```diff
@@ -1329,6 +1410,29 @@ MmNotPresentFaultSectionView(PMMSUPPORT AddressSpace,
                           &MemoryArea->Data.SectionData.RegionListHead,
                           Address, NULL);
     ASSERT(Region != NULL);
+
+    /* Check for a NOACCESS mapping */
+    if (Region->Protect & PAGE_NOACCESS)
+    {
+    	return STATUS_ACCESS_VIOLATION;
```
Comment (ThFabba) [inline ntoskrnl/mm/section.c]: Indentation is pretty broken in this whole hunk.

File: ntoskrnl/mm/section.c [hunk 2]
```diff
@@ -1275,6 +1275,87 @@ MiReadPage(PMEMORY_AREA MemoryArea,
 }
 #endif
 
+static VOID
```
Comment (ThFabba) [inline ntoskrnl/mm/section.c]: Did you make any changes to this function? Perhaps moving it should be a separate commit to make that clear.

Comment (zefklop) [reply ntoskrnl/mm/section.c]: No change. Ok for a separate commit.

File: ntoskrnl/mm/section.c (hunk 1 above)
Comment (zefklop) [reply ntoskrnl/mm/section.c]: Hmmm yeah. I will have to check the settings in my editor

File: ntoskrnl/mm/section.c (hunk 2 above)
Comment (zefklop) [reply ntoskrnl/mm/section.c]: Done

File: ntoskrnl/mm/section.c (hunk 1 above)
Comment (zefklop) [reply ntoskrnl/mm/section.c]: Now this should be fine.
